### ISSUE TITLE
Jobs lose their gitlab connections in the middle of builds

### ISSUE THREAD

**Issue by ace13** [Argelbargel/gitlab-branch-source-plugin#21]
Started running into a new issue now, when starting to deploy this onto our production Jenkins.

When a build runs, it seems to lose the GitLab connection property partway through. Filling the log up with messages to the tune of;
```
May 22, 2017 10:31:42 AM org.jenkinsci.plugins.workflow.job.WorkflowRun finish
INFO: GitLab/jenkins%2FWorkflow-modern/master #19 completed: FAILURE
May 22, 2017 10:31:42 AM argelbargel.jenkins.plugins.gitlab_branch_source.actions.GitLabSCMBuildStatusPublisher$Message run
WARNING: cannot publish build-status pending as no gitlab-connection is configured!
```

Worth noting is that if I set a connection property just before starting the build - while using build status reporting as `result`, then it manages to publish a status of 'running' on GitLab before the connection property disappears again.

I've tried sprinkling the pipelines with liberal amounts of
```groovy
properties([gitLabConnection('gitlab.fqdn')])
```
but they seem to still manage to lose the connection property before it can have any effect, both when publishing status for `stages` as well as `result`.

**Comment by ace13** [comment]
As a side-note, for my own API usage in the pipeline scripts I've been using something more along the lines of;
```groovy
def build = script.currentBuild.rawBuild 
def job = build.parent
def connection = job.parent.sources[0].source.connectionName

def api = Jenkins.instance.getDescriptor(com.dabsquared.gitlabjenkins.connection.GitLabConnectionConfig.class).getClient(connection)
```

**Comment by Argelbargel** [comment]
I've changed the publishing of build-statuses in 0.6.4 so that it does not rely on the GitLabConnectionProperty anymore. Hope this fixes your problem

**Comment by ace13** [comment]
Build status reporting indeed seems to do what it's supposed to now, many thanks.
